pages/[...slug].vue

```diff
@@ -0,0 +1,137 @@
+<script setup>
+// COMPONENT RE-IMPORTS
+// TODO: remove when we have implemented component library as a module
+// https://nuxt.com/docs/guide/directory-structure/components#library-authors
+import {
+  TwoColLayoutWStickySideBar, NavBreadcrumb, ResponsiveImage, CardMeta, RichText, PageAnchor, FlexibleBlocks, SectionWrapper,
+} from 'ucla-library-website-components'
+
+import { onMounted } from 'vue'
+
+// HELPERS
+import _get from 'lodash/get'
+
+// GQL
+import FTVA_GENERAL_CONTENT_DETAIL from '../gql/queries/FTVAGeneralContentDetail.gql'
+
+const { $graphql, $getHeaders } = useNuxtApp()
+
+const route = useRoute()
+
+const path = route.path.replace(/^\/|\/$/g, '') // trim initial and/or final slashes
+
+const variables = { path }
+
+const { data, error } = await useAsyncData(`general-content-${path}`, async () => {
+  const data = await $graphql.default.request(FTVA_GENERAL_CONTENT_DETAIL, variables)
+  return data
+})
+
+if (error.value) {
+  throw createError({
+    ...error.value, statusMessage: 'Page not found.' + error.value, fatal: true
+  })
+}
+
+if (!data.value.entry) {
+  // console.log('no data')
+  throw createError({
+    statusCode: 404,
+    statusMessage: 'Page Not Found',
+    fatal: true
+  })
+}
+
+if (data.value.entry.slug && process.server) {
+  const { $elasticsearchplugin } = useNuxtApp()
```

Comment:
do not use a plugin here use the composable we for ftva to index content, check other slug pages in ftva repo